dll/win32/browseui/shellbrowser.cpp

```diff
@@ -806,14 +806,22 @@ HRESULT CShellBrowser::BrowseToPIDL(LPCITEMIDLIST pidl, long flags)
     CComPtr<IShellFolder>                   newFolder;
     FOLDERSETTINGS                          newFolderSettings;
     HRESULT                                 hResult;
+    CLSID                                   clsid;
+    BOOL                                    HasIconViewType;
 
     // called by shell view to browse to new folder
     // also called by explorer band to navigate to new folder
     hResult = SHBindToFolder(pidl, &newFolder);
     if (FAILED_UNEXPECTEDLY(hResult))
         return hResult;
+    IUnknown_GetClassID(newFolder, &clsid);
```

Comment:
@cbialorucki Would you mind adding a comment that this is most likely a hack to get it working?

Reply:
Added a hack/fix me comment